microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/HelidonProxyServices.java

```diff
@@ -110,39 +133,75 @@ public boolean supportsClassDefining() {
 
         LOGGER.finest("Defining class " + className + " original class: " + originalClass.getName());
 
+        MethodHandles.Lookup lookup;
+
         try {
-            Module classModule = originalClass.getModule();
-            if (!myModule.canRead(classModule)) {
+            // lookup class name "guessed" from the class name of the proxy
+            String lookupClassName;
+
+            if (className.contains("$")) {
+                // package + first name in the compound proxy class name
+                lookupClassName = className.substring(0, className.indexOf('$'));
```

Comment:
`contains` effectively does an `indexOf`; no need to do `indexOf` again.  I'd rewrite this block something like this:
```
final int index = className.indexOf('$');
lookupClassName = index < 0 ? className : className.substring(0, index);
```

Reply:
Changed.